site/src/util/mrk.js

```diff
@@ -0,0 +1,128 @@
+const mrk = require('mrk.js')
+
+Object.assign(mrk.patterns, {
+  image({ read, readUntil }, meta) {
+    if (read(2) !== '![') return
+
+    // All characters up to `]` are the alt text
+    const alt = readUntil(']')
+
+    if (read(2) !== '](') return
+
+    // All characters up to `)` are the image src
+    const src = readUntil(')')
+
+    // Set metadata
+    meta({ alt, src })
+
+    return read() === ')'
+  },
+
+  // mrk supports 'code' already - but we don't want it to apply
+  // it if it's just '``' (i.e. no content)
+  code({ read, has }) {
+    if(read() === '`') {
+      if (read() === '`') return false
+
+      // Eat up every character until another backtick
+      let escaped = false, char, n
+
+      while (char = read()) {
+        if (char === '\\' && !escaped) escaped = true
+        else if (char === '`' && !escaped) return true
+        else escaped = false
+      }
+    }
+  },
+
+  codeblock({ read, readUntil, look }, meta) {
+    if (read(3) !== '```') return
+
+    let numBackticks = 3
+    while (look() === '`') {
+      numBackticks++
+      read()
+    }
+
+    // All characters up to newline following the intial
+    // set of backticks represent the language of the code
+    let lang = readUntil('\n')
+    read()
+
+    // Final fence
+    let code = ''
+    while (look(numBackticks) !== '`'.repeat(numBackticks)) {
+      if (look().length === 0) return false // We've reached the end
+      code += read()
+    }
+
+    read(numBackticks)
+    if (look() !== '\n' && look() !== '') return false
+
+    // Set metadata
+    meta({ lang, code })
+
+    return true
+  },
```

Comment:
I tried to implement codeblocks the other day and failed miserably. This looks nice! :tada:

Reply:
Was that what towerofnix/mrk was for?

Reply:
Yup. I think it was also to make some internal changes to support my thing (which is usually a bad idea anyways :P).